### PR TITLE
feat(Cell): swipe buttons have access to swipe methods

### PR DESCRIPTION
**Pull request by tormodAase**
This PR also introduces e2e testing with [detox](https://wix.github.io/Detox/) in Chronicles!

### CODE REVIEW

File: apps/chronicles/components/ComponentNavigationCell.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ export const ComponentNavigationCell = ({
     return (
         <Cell.Navigation
             title={title}
+            testID={title}
```
Comment (HavardNot) [inline apps/chronicles/components/ComponentNavigationCell.tsx:27]: Minor nitpick, should be renamed to `testId`

File: packages/components/src/components/_internal/SwipeableWithContext.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        throw new Error("Don't use legacy refs when referencing SwipeableWithContext");
+    const refToUse = ref ?? internalRef;
+    const swipeableMethods: SwipeableMethods = useMemo(
+        () => ({
+            close: logErrorIfUndefined(refToUse.current?.close),
+            openLeft: logErrorIfUndefined(refToUse.current?.openLeft),
+            openRight: logErrorIfUndefined(refToUse.current?.openRight),
+            reset: logErrorIfUndefined(refToUse.current?.reset),
+        }),
+        [refToUse, refToUse.current],
```
Comment (HavardNot) [inline packages/components/src/components/_internal/SwipeableWithContext.tsx:16]: you can remove the dependency to refToUse.current as mutable values won't cause the component to re-render

File: apps/chronicles/components/ComponentNavigationCell.tsx (hunk 1 above)
Comment (HavardNot) [reply apps/chronicles/components/ComponentNavigationCell.tsx:27]: I guess this property name was not defined in this pr so you can ignore this comment.

**Review by HavardNot — CHANGES_REQUESTED** [review]
looks like you got a warning from ESLint which should be rectified, other than that it looks good to me 🆒

File: apps/chronicles/components/ComponentNavigationCell.tsx (hunk 1 above)
Comment (tormodAase) [reply apps/chronicles/components/ComponentNavigationCell.tsx:27]: The property name is default in react, similar to `ref` and `key`

File: packages/components/src/components/_internal/SwipeableWithContext.tsx (hunk 2 above)
Comment (tormodAase) [reply packages/components/src/components/_internal/SwipeableWithContext.tsx]: Fixed

**Review by HavardNot — APPROVED** [review]

**Pull request merged**